packages/js/product-editor/src/products-app/index.tsx

```diff
@@ -13,12 +14,17 @@ import {
  * Internal dependencies
  */
 import { unlock } from '../lock-unlock';
+import useLayoutAreas from './router';
+import { Layout } from './layout';
 
 const { RouterProvider } = unlock( routerPrivateApis );
 const { GlobalStylesProvider } = unlock( editorPrivateApis );
 
 function ProductsLayout() {
-	return <div>Initial Products Layout</div>;
+	// This ensures the edited entity id and type are initialized properly.
+	useInitEditedEntityFromURL();
+	const route = useLayoutAreas();
+	return <Layout route={ route } />;
```

Comment:
Mimic the posts app implementation: https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/posts-app/index.js#L21